helper.go

```diff
@@ -463,24 +463,45 @@ func marshalObjectIdentifier(oid []int) (ret []byte, err error) {
 	return
 }
 
-func marshalOID(oid string) ([]byte, error) {
+func oidSplitToIntArray(oid string, sep string) ([]int, error) {
 	var err error
-
-	// Encode the oid
-	oid = strings.Trim(oid, ".")
-	oidParts := strings.Split(oid, ".")
-	oidBytes := make([]int, len(oidParts))
-
-	// Convert the string OID to an array of integers
-	for i := 0; i < len(oidParts); i++ {
-		oidBytes[i], err = strconv.Atoi(oidParts[i])
+	n := strings.Count(oid, sep) + 1
+	a := make([]int, n)
+	n--
+	i := 0
+	for i < n {
+		m := strings.Index(oid, sep)
+		if m < 0 {
+			break
+		}
+		a[i], err = strconv.Atoi(oid[:m])
```

Comment:
if an error occurs during ``strconv.Atoi(oid[:m])``, an error in the form of ``unable to marshal OID: unable to parse OID: value out of range`` is displayed by the calling ``marshalOID()`` function, which is a bit too verbose perhaps. Can we handle this in a nicer way?

Reply:
I also agree. Changed the error text